### PR TITLE
T248611 Restore "about" feature in article gallery view

### PR DESCRIPTION
**Pull request by hueitan**
Phabricator Link: https://phabricator.wikimedia.org/T248611

### Problem Statement

pcs `media` api is being removed recently, switch to mw api 

### Solution

fetch the api only when user press About center key

### Note

| About | Loading | No Image Info |
| --- | --- | --- |
| Screen Shot 2020-03-30 at 23 42 24 |Screen Shot 2020-03-30 at 23 48 23 | Screen Shot 2020-04-01 at 19 46 30
 |

### CODE REVIEW

**Comment by stephanebisson** [comment]
Article `en/Brazil`, 3rd image has no image info so the popup has only a header.

Screen Shot 2020-03-31 at 06 46 04

**Comment by medied** [comment]
Reviewing this PR was useful for me to further understand how to use `cachedFetch` and interact with APIs. I tested on Firefox and banana phone and I wasn't able to find any issues

File: src/components/Gallery.js [hunk 1]
```diff
@@ -30,45 +39,55 @@ const AboutContainer = ({ author, description, license, filePage, close }) => {
...
-            <p>
-              {author}{ author && <br /> }
-              {license}
-            </p>
-          </div>
-        )
-      }
+      <div>
+        <div class='sub-header'>{i18n('gallery-description')}</div>
+        <p class='description'>{mediaInfo.description || caption || title}</p>
```
Comment (medied) [inline src/components/Gallery.js:44]: We are always guaranteed to have at least `title` here, correct? That's why there's no need for a `gallery-unknown-description` label?

Comment (hueitan) [reply src/components/Gallery.js:44]: Yes

**Comment by hueitan** [comment]
this pr requires more update on the api, see the [comment in ticket](https://phabricator.wikimedia.org/T248611#6020386), in short summary, the image is not always from Commons.

**Comment by hueitan** [comment]
Ok, change is done and ready to be reviewed again, thanks for the input!

File: src/api/getArticleMediaList.js [hunk 2]
```diff
@@ -4,12 +4,14 @@ export const getArticleMediaList = (lang, title) => {
...
+      const source = item && item.srcset && `https:${item.srcset[0].src}`
       const media = {
         caption: item.caption && item.caption.text,
-        thumbnail: item && item.srcset && `https:${item.srcset[0].src}`,
+        thumbnail: source,
         title: item.title,
         canonicalizedTitle:
-          item.title && canonicalizeTitle(item.title.split(':')[1])
+          item.title && canonicalizeTitle(item.title.split(':')[1]),
+        fromCommon: source.indexOf('/commons') !== -1
```
Comment (medied) [inline src/api/getArticleMediaList.js:14]: Nice, good idea to declare `source` at the top

**Comment by medied** [comment]
Tested latest change on banana device using [Brasilien article](http://localhost:8080/#/article/de/Brasilien) and it worked as expected, btw

File: src/api/getArticleMediaInfo.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    return {
+      author,
+      description,
+      license: LicenseShortName && LicenseShortName.value,
+      filePage: imageInfo[0].descriptionshorturl
+    }
+  })
+}
+
+function strip (html) {
```
Comment (stephanebisson) [inline src/api/getArticleMediaInfo.js:38]: Could you make this `const strip = html => {...}` for consistency

Comment (stephanebisson) [reply src/api/getArticleMediaInfo.js]: I hate to be this picky... the next line has `var` but we prefer `const` or `let`

Comment (hueitan) [reply src/api/getArticleMediaInfo.js]: that's alright, i didn't pay too much attention when I copy this function from others, i should double check again, thanks !

**Pull request merged**